lib/backend/backend.go

```diff
@@ -123,19 +123,22 @@ func New(ctx context.Context, backend string, params Params) (Backend, error) {
 
 // IterateRange is a helper for stepping over a range
 func IterateRange(ctx context.Context, bk Backend, startKey []byte, endKey []byte, limit int, fn func([]Item) (stop bool, err error)) error {
+	if limit == 0 || limit > 10_000 {
+		limit = 10_000
+	}
 	for {
-		rslt, err := bk.GetRange(ctx, startKey, endKey, limit)
+		rslt, err := bk.GetRange(ctx, startKey, endKey, limit+1)
 		if err != nil {
 			return trace.Wrap(err)
 		}
-		stop, err := fn(rslt.Items)
+		stop, err := fn(rslt.Items[0:min(limit, len(rslt.Items))])
```

Comment:
If we plan to backport this to all branches (and I think we should), then we probably can't use `min` here.